edgecontext/timestamp.go

```diff
@@ -0,0 +1,74 @@
+package edgecontext
+
+import (
+	"encoding/json"
+	"strconv"
+	"time"
+)
+
+const millisecondsPerSecond = int64(time.Second / time.Millisecond)
+
+// TimestampMillisecond implements json encoding/decoding using milliseconds
+// since EPOCH.
+type TimestampMillisecond time.Time
+
+var (
+	_ json.Unmarshaler = (*TimestampMillisecond)(nil)
+	_ json.Marshaler   = TimestampMillisecond{}
+)
```

Comment:
🔕 Should we have a `lib` or `utils` package for more generic things like this?  There's not really a reason for it to be in `edgecontext`.

Reply:
Agreed, but I'll do that in a followup PR instead. We also have the microsecond types used in tracing that should be moved out as well.

Reply:
@pacejackson Not a big fan of generic packages, they become bit a dumping ground

Reply:
The package certainly shouldn't be named something too generic like `lib` or `utils`. I'm thinking about `timebp`.

Reply:
Yes. By the way, it seems the suffixing of `bp` is spreading across the repository. I find it makes the folder structure a lot harder to read. Wouldn't it be possible to just name those things what they are, i.e. `time` and if there's an import conflict, let the user rename it if necessary.

Reply:
most of those are the cases that importing conflict is bound to happen. for example metricsbp itself imported go-kit's metrics package, and timebp will certainly import stdlibs time library. I actually like the -bp suffix :)